mono/metadata/appdomain.c

```diff
@@ -2695,7 +2755,7 @@ mono_domain_try_unload (MonoDomain *domain, MonoObject **exc)
 
 	mono_domain_set (domain, FALSE);
 	/* Notify OnDomainUnload listeners */
-	method = mono_class_get_method_from_name (domain->domain->mbr.obj.vtable->klass, "DoDomainUnload", -1);	
```

Comment:
In time we should rename the old one as "old" and the new one as "regular" and maybe use ELF versioning to support old binaries. Unless we want to maintain binary compatibility with static linking, in which case, never change a name/signature.